### PR TITLE
Bump the Yams and swift-argument-parser tag for main and release/5.4

### PR DESCRIPTION
**Pull request by shahmishal**
* Yams (4.0.2) and swift-argument-parser (0.4.0) are required to support Apple Silicon

### CODE REVIEW

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
@swift-ci please test Apple Silicon macOS Platform

**Comment by shahmishal** [comment]
@swift-ci please test Apple Silicon macOS Platform

**Comment by shahmishal** [comment]
@swift-ci test

File: utils/update_checkout/update-checkout-config.json [hunk 1]
```diff
@@ -65,7 +65,7 @@
                 "llbuild": "main",
                 "swift-tools-support-core": "main",
                 "swiftpm": "main",
-                "swift-argument-parser": "0.3.0",
+                "swift-argument-parser": "0.4.0",
                 "swift-crypto": "1.1.4",
```
Comment (CodaFi) [inline utils/update_checkout/update-checkout-config.json:69]: We have to bump this to `1.1.5` too

File: utils/update_checkout/update-checkout-config.json [hunk 2]
```diff
@@ -124,7 +124,7 @@
...
```
Comment (CodaFi) [inline utils/update_checkout/update-checkout-config.json:128]: And here

File: utils/update_checkout/update-checkout-config.json (hunk 1 above)
Comment (shahmishal) [reply utils/update_checkout/update-checkout-config.json:69]: @yim-lee created another PR for it.

Comment (yim-lee) [reply utils/update_checkout/update-checkout-config.json:69]: https://github.com/apple/swift/pull/36314

**Pull request merged**